plugin/trino-hudi/src/main/java/io/trino/plugin/hudi/HudiSplitSource.java

```diff
@@ -151,4 +190,40 @@ public static HudiSplitWeightProvider createSplitWeightProvider(ConnectorSession
         }
         return HudiSplitWeightProvider.uniformStandardWeightProvider();
     }
+
+    static boolean partitionMatchesPredicate(
+            HudiSplit split,
+            TupleDomain<HiveColumnHandle> dynamicFilterPredicate)
+    {
+        if (dynamicFilterPredicate.isNone()) {
+            return false;
+        }
+
+        Map<HiveColumnHandle, Domain> domains = dynamicFilterPredicate.getDomains().orElseThrow();
+
+        // Match each partition key against its domain in the dynamic filter
+        for (HivePartitionKey partitionKey : split.getPartitionKeys()) {
+            for (Map.Entry<HiveColumnHandle, Domain> entry : domains.entrySet()) {
+                HiveColumnHandle column = entry.getKey();
+                if (!column.isPartitionKey() || !column.getName().equals(partitionKey.name())) {
+                    continue;
+                }
+
+                Domain domain = entry.getValue();
+                NullableValue value = HiveUtil.getPrefilledColumnValue(
+                        column,
+                        partitionKey,
+                        null,  // path
+                        OptionalInt.empty(), // bucketNumber
+                        0,    // fileSize
+                        0,    // fileModifiedTime
+                        ""); // partitionName
```

Comment:
Do these matter?  Right now they are placeholder values.

Reply:
I'll refactor this part to reduce the number of NullableValue instantiation and reduce complexity using a Map.

Reply:
Got it.  We can address this in a follow-up.